### ISSUE TITLE
Annotations & Feedback - Sprint 4

### ISSUE THREAD

**Issue by ducnguyen67201** [ducnguyen67201/tesingrepodkjsldfjisjslkdnsdvnsdlvsd#105]
## Overview

Build annotation and feedback capabilities to collect human input on traces. This enables fine-tuning data collection, QA workflows, and team collaboration on quality assessment.

**Points:** 5
**Priority:** P0
**Dependencies:** #104 Evaluations & Scoring (optional)

---

## Use Cases

| Use Case | Example |
|----------|---------|
| **Fine-tuning data** | Collect corrections for training data |
| **QA workflows** | Team reviews and approves responses |
| **End-user feedback** | Capture thumbs up/down from users |
| **Labeling** | Tag outputs with categories |
| **Notes** | Add context for debugging |

---

### Data Model

\`\`\`prisma
model Annotation {
  id          String          @id @default(cuid())
  projectId   String
  project     Project         @relation(fields: [projectId], references: [id], onDelete: Cascade)
  traceId     String?
  trace       Trace?          @relation(fields: [traceId], references: [id], onDelete: Cascade)
  spanId      String?
  span        Span?           @relation(fields: [spanId], references: [id], onDelete: Cascade)
  
  type        AnnotationType  // CORRECTION, LABEL, NOTE, FEEDBACK
  content     String          // The annotation text or JSON
  rating      Int?            // -1 (thumbs down), 0 (neutral), 1 (thumbs up)
  
  annotatorId String?         // Dashboard user
  annotator   User?           @relation(fields: [annotatorId], references: [id])
  
  // For end-user feedback (via SDK)
  trackedUserId String?
  trackedUser   TrackedUser?  @relation(fields: [trackedUserId], references: [id])
  
  metadata    Json?
  createdAt   DateTime        @default(now())
  updatedAt   DateTime        @updatedAt
  
  @@index([projectId, createdAt(sort: Desc)])
  @@index([traceId])
  @@index([spanId])
  @@index([type, projectId])
  @@map("annotations")
}

enum AnnotationType {
  CORRECTION  // Fixed/improved response
  LABEL       // Category tag
  NOTE        // Context/debugging note
  FEEDBACK    // End-user feedback
}
\`\`\`

---

## Deliverables

| Component | Type | Priority |
|-----------|------|----------|
| Annotation model | Migration | P0 |
| AnnotationType enum | Migration | P0 |
| Annotations tRPC router | API | P0 |
| Annotation Zod schemas | API | P0 |
| SDK feedback() method | SDK | P0 |
| Annotation panel in trace detail | Web | P1 |
| Annotation list view | Web | P1 |
| Export annotations as JSONL | API | P2 |

---

## API Endpoints

\`\`\`typescript
// packages/api/src/routers/annotations.ts
annotationsRouter = {
  create: // Create annotation
  list: // List with filters (type, project, user)
  get: // Get single annotation
  update: // Update content/rating
  delete: // Remove annotation
  byTrace: // Get all annotations for a trace
  export: // Export as JSONL for fine-tuning
}
\`\`\`

---

## SDK Usage

\`\`\`typescript
// End-user feedback (thumbs up/down)
CognObserve.feedback({
  traceId: 'trace-123',
  rating: 1, // thumbs up
  comment: 'This was helpful!',
});

// Within observe context
await CognObserve.observe('chat', async () => {
  const response = await llm.generate({...});
  
  // Collect feedback later via callback
  onUserFeedback((rating) => {
    CognObserve.feedback({ rating });
  });
  
  return response;
});
\`\`\`

---

## UI Components

\`\`\`
apps/web/src/
├── components/
│   └── annotations/
│       ├── annotation-panel.tsx      # Side panel for adding annotations
│       ├── annotation-list.tsx       # List of annotations
│       ├── annotation-form.tsx       # Form for creating/editing
│       ├── rating-buttons.tsx        # Thumbs up/down component
│       └── correction-diff.tsx       # Show original vs corrected
\`\`\`

---

## Export Format

\`\`\`jsonl
{"prompt": "...", "completion": "...", "correction": "...", "rating": 1}
{"prompt": "...", "completion": "...", "labels": ["factual", "concise"]}
\`\`\`

---

## Definition of Done

- [ ] Database migration applied
- [ ] Annotation model with indexes
- [ ] Annotations tRPC router with all endpoints
- [ ] Zod schemas for validation
- [ ] SDK feedback() method implemented
- [ ] Annotation panel in trace detail view
- [ ] Rating buttons (thumbs up/down)
- [ ] Export to JSONL endpoint
- [ ] Unit tests passing
- [ ] Integration tests passing
